refinery/core/utils.py

```diff
@@ -1031,3 +1016,14 @@ def _wrapped_view(request, *args, **kwargs):
                 request.recaptcha_is_valid = True
         return view_function(request, *args, **kwargs)
     return _wrapped_view
+
+
+def get_data_set_for_view_set(uuid):
+    try:
+        return core.models.DataSet.objects.get(uuid=uuid)
+    except core.models.DataSet.DoesNotExist as e:
+        logger.error(e)
+        raise Http404
+    except core.models.DataSet.MultipleObjectsReturned as e:
+        logger.error(e)
+        raise APIException("Multiple dataSets returned for this request.")
```

Comment:
Might be worth simplifying this method using `get_object_or_404()` shortcut: https://stackoverflow.com/questions/22552306/what-will-happen-if-get-object-or-404-finds-many-objects (also, this pattern might be applicable in other similar cases)